openmc/filter.py

```diff
@@ -1610,3 +1610,222 @@ class DelayedGroupFilter(IntegralFilter):
+        # Make sure the values are in a numpy array.
+        y = np.array(y)
+
+        # If the array is 0D, promote to 1D.
+        if y.shape == ():
+            y.shape = (1,)
+
+        # Make sure the values are Real.
+        cv.check_type('filter interpolant values', y, Iterable, Real)
+
+        self._y = y
+
+    @bins.setter
+    def bins(self, bins):
+        raise RuntimeError('EnergyFunctionFilters have no bins.')
+
+    def to_xml(self):
+        """Return XML Element representing the Filter."""
+        element = ET.Element('filter')
+        element.set('type', self.short_name.lower())
+        element.set('energy', ' '.join(str(e) for e in self.energy))
+        element.set('y', ' '.join(str(y) for y in self.y))
+        return element
+
+    def can_merge(self, other):
+        return False
+
+    def is_subset(self, other):
+        return self == other
+
+    def get_bin_index(self, filter_bin):
+        # This filter only has one bin.  Always return 0.
+        return 0
+
+    def get_bin(self, bin_index):
+        """This function is invalid for EnergyFunctionFilters."""
+        raise RuntimeError('EnergyFunctionFilters have no get_bin() method')
+
+    def get_pandas_dataframe(self, data_size, **kwargs):
+        """Builds a Pandas DataFrame for the Filter's bins.
+
+        This method constructs a Pandas DataFrame object for the filter with
+        columns annotated by filter bin information. This is a helper method for
+        :meth:`Tally.get_pandas_dataframe`.
+
+        Parameters
+        ----------
+        data_size : Integral
+            The total number of bins in the tally corresponding to this filter
+
+        Returns
+        -------
+        pandas.DataFrame
+            A Pandas DataFrame with a column that is filled with a hash of this
+            filter. EnergyFunctionFilters have only 1 bin so the purpose of this
+            DataFrame column is to differentiate the filter from other
+            EnergyFunctionFilters. The number of rows in the DataFrame is the
+            same as the total number of bins in the corresponding tally.
+
+        Raises
+        ------
+        ImportError
+            When Pandas is not installed
+
+        See also
+        --------
+        Tally.get_pandas_dataframe(), CrossFilter.get_pandas_dataframe()
+
+        """
+
+        import pandas as pd
+        df = pd.DataFrame()
+
+        # There is no clean way of sticking all the energy, y data into a
+        # DataFrame so instead we'll just make a column with the filter name
+        # and fill it with a hash of the __repr__.  We want a hash that is
+        # reproducible after restarting the interpreter so we'll use hashlib.md5
+        # rather than the intrinsic hash().
+        from hashlib import md5
+        hash_fun = md5()
+        hash_fun.update(repr(self).encode('utf-8'))
+        out = hash_fun.hexdigest()
+
+        # The full 16 bytes make for a really wide column.  Just 7 bytes of the
```

Comment:
2 hex characters = 16^2 bits = 256 bits = 1 byte.  I'll change the comment to make this more clear though